test/brsTypes/components/ArrayGrid.test.js

```diff
@@ -9,27 +9,27 @@ describe("ArrayGrid", () => {
             expect(group.toString()).toEqual(
                 `<Component: roSGNode:ArrayGrid> =
 {
-    change: <UNINITIALIZED>
+    change: <Component: roAssociativeArray>
     focusable: false
     focusedchild: invalid
     id: 
     visible: true
     opacity: 1
-    translation: invalid
+    translation: <Component: roArray>
     rotation: 0
-    scale: invalid
-    scalerotatecenter: invalid
+    scale: <Component: roArray>
+    scalerotatecenter: <Component: roArray>
```

Comment:
For all of these components, how did you know which of these members were of type `<Compenent: roArray>`?

Reply:
Good question! In a component definition, we have a list of default fields, and each field has a `type` (see [here](https://github.com/sjbarag/brs/blob/master/src/brsTypes/components/Group.ts#L8) for an example). That `type` corresponds to the [Roku documentation](https://developer.roku.com/docs/references/scenegraph/layout-group-nodes/group.md) for whichever component we're creating.